app/jobs/psql_stats_job.rb

```diff
@@ -0,0 +1,81 @@
+  FROM (
+    SELECT ceil( reltuples / ( (bs-page_hdr)/tpl_size ) ) + ceil( toasttuples / 4 ) AS est_tblpages,
+      ceil( reltuples / ( (bs-page_hdr)*fillfactor/(tpl_size*100) ) ) + ceil( toasttuples / 4 ) AS est_tblpages_ff,
+      tblpages, fillfactor, bs, tblid, schemaname, tblname, heappages, toastpages, is_na
+      -- , tpl_hdr_size, tpl_data_size, pgstattuple(tblid) AS pst -- (DEBUG INFO)
+    FROM (
+      SELECT
+        ( 4 + tpl_hdr_size + tpl_data_size + (2*ma)
+          - CASE WHEN tpl_hdr_size%ma = 0 THEN ma ELSE tpl_hdr_size%ma END
+          - CASE WHEN ceil(tpl_data_size)::int%ma = 0 THEN ma ELSE ceil(tpl_data_size)::int%ma END
+        ) AS tpl_size, bs - page_hdr AS size_per_block, (heappages + toastpages) AS tblpages, heappages,
+        toastpages, reltuples, toasttuples, bs, page_hdr, tblid, schemaname, tblname, fillfactor, is_na
+        -- , tpl_hdr_size, tpl_data_size
+      FROM (
+        SELECT
+          tbl.oid AS tblid, ns.nspname AS schemaname, tbl.relname AS tblname, tbl.reltuples,
+          tbl.relpages AS heappages, coalesce(toast.relpages, 0) AS toastpages,
+          coalesce(toast.reltuples, 0) AS toasttuples,
+          coalesce(substring(
+            array_to_string(tbl.reloptions, ' ')
+            FROM 'fillfactor=([0-9]+)')::smallint, 100) AS fillfactor,
+          current_setting('block_size')::numeric AS bs,
+          CASE WHEN version()~'mingw32' OR version()~'64-bit|x86_64|ppc64|ia64|amd64' THEN 8 ELSE 4 END AS ma,
+          24 AS page_hdr,
+          23 + CASE WHEN MAX(coalesce(s.null_frac,0)) > 0 THEN ( 7 + count(s.attname) ) / 8 ELSE 0::int END
+            + CASE WHEN bool_or(att.attname = 'oid' and att.attnum < 0) THEN 4 ELSE 0 END AS tpl_hdr_size,
+          sum( (1-coalesce(s.null_frac, 0)) * coalesce(s.avg_width, 0) ) AS tpl_data_size,
+          bool_or(att.atttypid = 'pg_catalog.name'::regtype)
+            OR sum(CASE WHEN att.attnum > 0 THEN 1 ELSE 0 END) <> count(s.attname) AS is_na
+        FROM pg_attribute AS att
+          JOIN pg_class AS tbl ON att.attrelid = tbl.oid
+          JOIN pg_namespace AS ns ON ns.oid = tbl.relnamespace
+          LEFT JOIN pg_stats AS s ON s.schemaname=ns.nspname
+            AND s.tablename = tbl.relname AND s.inherited=false AND s.attname=att.attname
+          LEFT JOIN pg_class AS toast ON tbl.reltoastrelid = toast.oid
+          WHERE NOT att.attisdropped AND ns.nspname='public' -- remove `AND ns.nspname='public'` to include system namespaces
+          AND tbl.relkind in ('r','m')
+        GROUP BY 1,2,3,4,5,6,7,8,9,10
+        ORDER BY 2,3
+      ) AS s
+    ) AS s2
+  ) AS s3
+  -- WHERE NOT is_na
+  --   AND tblpages*((pst).free_percent + (pst).dead_tuple_percent)::float4/100 >= 1
+  ORDER BY schemaname, tblname;
+  SQL
+
+  def perform
+    ar_result = ActiveRecord::Base.connection.exec_query(QUERY)
+    IdentityJobLogSubscriber.new.logger.info(
```

Comment:
I think we can do this instead of instantiating
```suggestion
    IdentityJobLogSubscriber.worker_logger.info(
```

Reply:
Another option is to use the report logger?

```suggestion
    IdentityJobLogSubscriber.reports_logger.info(
```

Reply:
What's the difference? I only need the data written to a log file so we can ingest in cloudwatch.

Reply:
the reports log file is for the results of reports, so it probably fits better here

worker log file is currently where GoodJob/ActiveJob metrics go

Reply:
Ahh that  makes sense, I'll move it to the reports_logger. Thanks